# Fix makefile to use EB environment
# Ward Poelmans, Alex Domingo (Vrije Universiteit Brussel)
--- makefile.orig	2020-04-18 01:53:22.403951000 +0200
+++ makefile	2020-04-18 01:52:16.444386000 +0200
@@ -35,7 +35,7 @@
 INCLUDEDIR = /usr/local/include
 LIBDIR     = /usr/local/lib
 
-CFLAGS     = -O3 -Wall
+CFLAGS     = $(OPTFLAGS) -Wall
 SHLIB_CFLAGS = -fPIC
 mpicxx=mpic++
 
@@ -48,28 +48,28 @@
 MPLRSOBJ64=lrslong1-mplrs.o lrslib1-mplrs.o lrslibgmp-mplrs.o lrsgmp-mplrs.o lrsdriver.o mplrs64.o
 
 lrs: ${LRSOBJ}
-	$(CC) ${CFLAGS} -DMA -DB128 -L${LIBDIR} -o lrs ${LRSOBJ} -lgmp
+	$(CC) ${CFLAGS} -DMA -DB128 $(LDFLAGS) -o lrs ${LRSOBJ} -lgmp
 
 lrs64: ${LRSOBJ64}
-	$(CC) ${CFLAGS} -DMA -L${LIBDIR} -o lrs ${LRSOBJ64} -lgmp
+	$(CC) ${CFLAGS} -DMA $(LDFLAGS) -o lrs ${LRSOBJ64} -lgmp
 
 redund: ${REDUNDOBJ}
-	$(CC) ${CFLAGS} -DMA -DB128 -L${LIBDIR} -o redund ${REDUNDOBJ} -lgmp
+	$(CC) ${CFLAGS} -DMA -DB128 $(LDFLAGS) -o redund ${REDUNDOBJ} -lgmp
 
 redund64: ${REDUNDOBJ64}
-	$(CC) ${CFLAGS} -DMA -L${LIBDIR} -o redund ${REDUNDOBJ64} -lgmp
+	$(CC) ${CFLAGS} -DMA $(LDFLAGS) -o redund ${REDUNDOBJ64} -lgmp
 
 lrs.o: lrs.c
-	$(CC) ${CFLAGS} -DMA -DB128 -L${LIBDIR} -c -o lrs.o lrs.c
+	$(CC) ${CFLAGS} -DMA -DB128 $(LDFLAGS) -c -o lrs.o lrs.c
 
 lrs64.o: lrs.c
-	$(CC) ${CFLAGS} -DMA -L${LIBDIR} -c -o lrs64.o lrs.c
+	$(CC) ${CFLAGS} -DMA $(LDFLAGS) -c -o lrs64.o lrs.c
 
 redund.o: redund.c
-	$(CC) ${CFLAGS} -DMA -DB128 -L${LIBDIR} -c -o redund.o redund.c
+	$(CC) ${CFLAGS} -DMA -DB128 $(LDFLAGS) -c -o redund.o redund.c
 
 redund64.o: redund.c
-	$(CC) ${CFLAGS} -DMA -L${LIBDIR} -c -o redund64.o redund.c
+	$(CC) ${CFLAGS} -DMA $(LDFLAGS) -c -o redund64.o redund.c
 
 lrslong1.o: lrslong.c lrslong.h
 	$(CC) ${CFLAGS} -DMA -DSAFE -DLRSLONG -c -o lrslong1.o lrslong.c
@@ -84,10 +84,10 @@
 	$(CC) ${CFLAGS} -DMA -DSAFE -DB128 -DLRSLONG -c -o lrslib2.o lrslib.c
 
 lrslibgmp.o: lrslib.c lrslib.h
-	$(CC) ${CFLAGS} -DMA -DGMP -I${INCLUDEDIR} -c -o lrslibgmp.o lrslib.c
+	$(CC) ${CFLAGS} -DMA -DGMP $(CPPFLAGS) -c -o lrslibgmp.o lrslib.c
 
 lrsgmp.o: lrsgmp.c lrsgmp.h
-	$(CC) ${CFLAGS} -DMA -DGMP -I${INCLUDEDIR} -c -o lrsgmp.o lrsgmp.c
+	$(CC) ${CFLAGS} -DMA -DGMP $(CPPFLAGS) -c -o lrsgmp.o lrsgmp.c
 
 
 lrslong1-mplrs.o: lrslong.c lrslong.h
@@ -103,25 +103,25 @@
 	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -DMA -DSAFE -DB128 -DLRSLONG -DPLRS -c -o lrslib2-mplrs.o lrslib.c
 
 lrslibgmp-mplrs.o: lrslib.c lrslib.h
-	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS -I${INCLUDEDIR} -c -o lrslibgmp-mplrs.o lrslib.c
+	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS $(CPPFLAGS) -c -o lrslibgmp-mplrs.o lrslib.c
 
 lrsgmp-mplrs.o: lrsgmp.c lrsgmp.h
-	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS -I${INCLUDEDIR} -c -o lrsgmp-mplrs.o lrsgmp.c
+	$(mpicxx) ${CFLAGS} -DMA -DTIMES -DSIGNALS -DGMP -DPLRS $(CPPFLAGS) -c -o lrsgmp-mplrs.o lrsgmp.c
 
 mplrs.o: mplrs.c mplrs.h lrslib.h lrsgmp.h
-	$(mpicxx) ${CFLAGS} -I${INCLUDEDIR} -DMA -DPLRS -DTIMES -DB128 -DSIGNALS -D_WITH_GETLINE -c -o mplrs.o mplrs.c
+	$(mpicxx) ${CFLAGS} $(CPPFLAGS) -DMA -DPLRS -DTIMES -DB128 -DSIGNALS -D_WITH_GETLINE -c -o mplrs.o mplrs.c
 
 mplrs64.o: mplrs.c mplrs.h lrslib.h lrsgmp.h
-	$(mpicxx) ${CFLAGS} -I${INCLUDEDIR} -DMA -DPLRS -DTIMES -DSIGNALS -D_WITH_GETLINE -c -o mplrs64.o mplrs.c
+	$(mpicxx) ${CFLAGS} $(CPPFLAGS) -DMA -DPLRS -DTIMES -DSIGNALS -D_WITH_GETLINE -c -o mplrs64.o mplrs.c
 
 mplrs: ${MPLRSOBJ} mplrsgmp
-	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -static-libstdc++ -Wno-write-strings -Wno-sign-compare -DPLRS -DMA -DB128 -L${LIBDIR} -o mplrs ${MPLRSOBJ} -lgmp
+	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -static-libstdc++ -Wno-write-strings -Wno-sign-compare -DPLRS -DMA -DB128 $(LDFLAGS) -o mplrs ${MPLRSOBJ} -lgmp
 
 mplrs64: ${MPLRSOBJ64} mplrsgmp
-	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -static-libstdc++ -Wno-write-strings -Wno-sign-compare -DPLRS -DMA -L${LIBDIR} -o mplrs ${MPLRSOBJ64} -lgmp
+	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -static-libstdc++ -Wno-write-strings -Wno-sign-compare -DPLRS -DMA $(LDFLAGS) -o mplrs ${MPLRSOBJ64} -lgmp
 
 mplrsgmp: mplrs.c mplrs.h lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c
-	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -static-libstdc++ -DPLRS -DGMP -I${INCLUDEDIR} mplrs.c lrslib.c lrsgmp.c lrsdriver.c -L${LIBDIR} -o mplrsgmp -lgmp
+	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -static-libstdc++ -DPLRS -DGMP $(CPPFLAGS) mplrs.c lrslib.c lrsgmp.c lrsdriver.c $(LDFLAGS) -o mplrsgmp -lgmp
 
 mplrs1: mplrs.c mplrs.h lrslib.c lrslib.h lrslong.c lrslong.h lrsdriver.h lrsdriver.c
 	$(mpicxx) ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -static-libstdc++ -DPLRS -DSAFE -DLRSLONG mplrs.c lrslib.c lrslong.c lrsdriver.c -o mplrs1
@@ -135,59 +135,59 @@
 
 flint:	 	lrs.c lrslib.c lrslib.h lrsgmp.c lrsgmp.h
 		@test -d  ${INCLUDEDIR}/flint || { echo ${INCLUDEDIR}/flint not found; exit 1; }
-		$(CC) -O3 -DFLINT -I${INCLUDEDIR} -I${INCLUDEDIR}/flint lrs.c lrsdriver.c lrslib.c lrsgmp.c -L${LIBDIR} -lflint -o lrsflint -lgmp
+		$(CC) $(OPTFLAGS) -DFLINT $(CPPFLAGS) $(CPPFLAGS)/flint lrs.c lrsdriver.c lrslib.c lrsgmp.c $(LDFLAGS) -lflint -o lrsflint -lgmp
 
 mplrsflint:	mplrs.c mplrs.h lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.c lrsdriver.h
-	${mpicxx} ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DFLINT -Wno-write-strings -Wno-sign-compare -I${INCLUDEDIR}/flint -DPLRS -o mplrsflint mplrs.c lrsdriver.c lrslib.c lrsgmp.c -L${LIBDIR} -lflint -lgmp
+	${mpicxx} ${CFLAGS} -DTIMES -DSIGNALS -D_WITH_GETLINE -DFLINT -Wno-write-strings -Wno-sign-compare $(CPPFLAGS)/flint -DPLRS -o mplrsflint mplrs.c lrsdriver.c lrslib.c lrsgmp.c $(LDFLAGS) -lflint -lgmp
 
 #comment out lines with -DB128 if __int128 not supported by your C compiler
 
 lrsgmp:		lrs.c lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c redund.c
-		gcc  -O3   -DGMP -I${INCLUDEDIR} -o lrsgmp lrs.c lrslib.c lrsgmp.c lrsdriver.c -L${LIBDIR}  -lgmp
-		$(CC)  -O3  -DGMP -I${INCLUDEDIR}  redund.c lrslib.c lrsgmp.c lrsdriver.c -L${LIBDIR} -lgmp -o redundgmp
+		gcc  $(OPTFLAGS)   -DGMP $(CPPFLAGS) -o lrsgmp lrs.c lrslib.c lrsgmp.c lrsdriver.c $(LDFLAGS)  -lgmp
+		$(CC)  $(OPTFLAGS)  -DGMP $(CPPFLAGS)  redund.c lrslib.c lrsgmp.c lrsdriver.c $(LDFLAGS) -lgmp -o redundgmp
 
 single:		lrs.c lrslong.c lrslong.h lrslib.c lrslib.h lrsgmp.c lrsgmp.h lrsdriver.h lrsdriver.c
-		$(CC)  -O3 -DSAFE  -DLRSLONG -o lrs1 lrs.c lrslib.c lrslong.c lrsdriver.c
-		$(CC)  -O3 -DLRSLONG -o lrs1n lrs.c lrslib.c lrslong.c lrsdriver.c
-		$(CC)  -O3 -DB128 -DSAFE  -DLRSLONG -o lrs2 lrs.c lrslib.c lrslong.c lrsdriver.c
-		$(CC)  -O3 -DB128 -DLRSLONG -o lrs2n lrs.c lrslib.c lrslong.c lrsdriver.c
-		$(CC)  -O3 -DSAFE  -DLRSLONG redund.c lrslib.c lrslong.c lrsdriver.c -o redund1
-		$(CC)  -O3 -DB128 -DSAFE  -DLRSLONG -DB128 redund.c lrslib.c lrslong.c lrsdriver.c -o redund2
+		$(CC)  $(OPTFLAGS) -DSAFE  -DLRSLONG -o lrs1 lrs.c lrslib.c lrslong.c lrsdriver.c
+		$(CC)  $(OPTFLAGS) -DLRSLONG -o lrs1n lrs.c lrslib.c lrslong.c lrsdriver.c
+		$(CC)  $(OPTFLAGS) -DB128 -DSAFE  -DLRSLONG -o lrs2 lrs.c lrslib.c lrslong.c lrsdriver.c
+		$(CC)  $(OPTFLAGS) -DB128 -DLRSLONG -o lrs2n lrs.c lrslib.c lrslong.c lrsdriver.c
+		$(CC)  $(OPTFLAGS) -DSAFE  -DLRSLONG redund.c lrslib.c lrslong.c lrsdriver.c -o redund1
+		$(CC)  $(OPTFLAGS) -DB128 -DSAFE  -DLRSLONG -DB128 redund.c lrslib.c lrslong.c lrsdriver.c -o redund2
 		make lrsgmp
 		make redundgmp
 
 		@test -d  ${INCLUDEDIR}/flint || { echo ${INCLUDEDIR}/flint not found: lrsflint not created; exit 1; }
-		$(CC)  -O3  -DFLINT -I${INCLUDEDIR}/flint lrs.c lrslib.c lrsgmp.c lrsdriver.c -L${LIBDIR} -lflint -o lrsflint -lgmp
-		$(CC)  -O3  -DFLINT -I${INCLUDEDIR}/flint redund.c lrslib.c lrsgmp.c lrsdriver.c -L${LIBDIR}  -lflint -o redundflint -lgmp
+		$(CC)  $(OPTFLAGS)  -DFLINT $(CPPFLAGS)/flint lrs.c lrslib.c lrsgmp.c lrsdriver.c $(LDFLAGS) -lflint -o lrsflint -lgmp
+		$(CC)  $(OPTFLAGS)  -DFLINT $(CPPFLAGS)/flint redund.c lrslib.c lrsgmp.c lrsdriver.c $(LDFLAGS)  -lflint -o redundflint -lgmp
 
 
 
 allmp:		lrs.c lrslib.c lrslib.h lrsmp.c lrsmp.h lrsdriver.h lrsdriver.c
-		$(CC) -Wall -O3  -o lrs lrs.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -Wall -O3  -DLRSLONG -o lrs1 lrs.c lrslib.c lrsdriver.c lrslong.c
-		$(CC) -O3  -o redund  redund.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3  -DLRSLONG -o redund1  redund.c lrslib.c lrsdriver.c lrslong.c
-		$(CC) -O3 -DLRS_QUIET   -o lrsnash lrsnash.c lrsnashlib.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3  -o setnash setupnash.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3  -o setnash2 setupnash2.c lrslib.c lrsdriver.c lrsmp.c
-		$(CC) -O3  -o 2nash 2nash.c
+		$(CC) -Wall $(OPTFLAGS)  -o lrs lrs.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) -Wall $(OPTFLAGS)  -DLRSLONG -o lrs1 lrs.c lrslib.c lrsdriver.c lrslong.c
+		$(CC) $(OPTFLAGS)  -o redund  redund.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS)  -DLRSLONG -o redund1  redund.c lrslib.c lrsdriver.c lrslong.c
+		$(CC) $(OPTFLAGS) -DLRS_QUIET   -o lrsnash lrsnash.c lrsnashlib.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS)  -o setnash setupnash.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS)  -o setnash2 setupnash2.c lrslib.c lrsdriver.c lrsmp.c
+		$(CC) $(OPTFLAGS)  -o 2nash 2nash.c
 
 demo:	lpdemo1.c lrslib.c lrsdriver.c lrslib.h lrsgmp.c lrsgmp.h
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lpdemo1 lpdemo1.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lpdemo lpdemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lpdemo2 lpdemo2.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
-	$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o vedemo  vedemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lpdemo1 lpdemo1.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lpdemo lpdemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lpdemo2 lpdemo2.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
+	$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o vedemo  vedemo.c lrslib.c lrsdriver.c lrsgmp.c -lgmp -DGMP
 
 lrsnash:	lrsnash.c nashdemo.c lrsnashlib.c lrslib.c lrsnashlib.h lrslib.h lrsgmp.c lrsgmp.h lrslong.h lrsdriver.h lrsdriver.c
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnashgmp lrsnash.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c  -lgmp -DGMP
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash1 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o lrsnash2 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE -DB128
-		$(CC) -O3   -I${INCLUDEDIR} -L${LIBDIR} -o nashdemo nashdemo.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c -lgmp -DGMP
-		$(CC) -O3  -I${INCLUDEDIR} -L${LIBDIR} -o 2nash 2nash.c
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lrsnashgmp lrsnash.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c  -lgmp -DGMP
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lrsnash1 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o lrsnash2 lrsnash.c lrsnashlib.c lrslib.c lrslong.c lrsdriver.c -DLRSLONG -DSAFE -DB128
+		$(CC) $(OPTFLAGS)   $(CPPFLAGS) $(LDFLAGS) -o nashdemo nashdemo.c lrsnashlib.c lrslib.c lrsgmp.c lrsdriver.c -lgmp -DGMP
+		$(CC) $(OPTFLAGS)  $(CPPFLAGS) $(LDFLAGS) -o 2nash 2nash.c
 		cp lrsnashgmp lrsnash
 
 fourier:	fourier.c lrslib.h lrslib.c lrsgmp.h lrsgmp.c
-	$(CC) -O3   -DGMP -I${INCLUDEDIR} fourier.c lrslib.c lrsdriver.c lrsgmp.c -L${LIBDIR}  -lgmp -o fourier
+	$(CC) $(OPTFLAGS)   -DGMP $(CPPFLAGS) fourier.c lrslib.c lrsdriver.c lrsgmp.c $(LDFLAGS)  -lgmp -o fourier
 
 ######################################################################
 # From here on the author is David Bremner <bremner@unb.ca> to whom you should turn for help             
@@ -201,7 +201,7 @@
 SHLIBOBJ=lrslong1-shr.o lrslong2-shr.o lrslib1-shr.o lrslib2-shr.o \
 	lrslibgmp-shr.o lrsgmp-shr.o lrsdriver-shr.o
 
-SHLIBBIN=lrs-shared redund-shared lrsnash-shared
+SHLIBBIN=lrs-shared redund-shared lrsnash-shared mplrs lrsgmp mplrsgmp redundgmp
 
 # Building (linking) the shared library, and relevant symlinks.
 
@@ -225,7 +225,7 @@
 	$(CC) redund.o -o $@ -L . -llrs
 
 lrsnash-shared: ${SHLINK}  lrsnash.c
-	$(CC) -DGMP -DMA lrsnash.c  lrsnashlib.c -I${INCLUDEDIR} -o $@ -L . -llrs -lgmp
+	$(CC) -DGMP -DMA lrsnash.c  lrsnashlib.c $(CPPFLAGS) -o $@ -L . -llrs -lgmp
 
 # build object files for the shared library
 
@@ -242,10 +242,10 @@
 	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -DMA -DSAFE -DB128 -DLRSLONG -c -o $@ lrslong.c
 
 lrslibgmp-shr.o: lrslib.c lrslib.h
-	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -DMA -DGMP -I${INCLUDEDIR} -c -o $@ lrslib.c
+	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -DMA -DGMP $(CPPFLAGS) -c -o $@ lrslib.c
 
 lrsgmp-shr.o: lrsgmp.c lrsgmp.h
-	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -DMA -DGMP -I${INCLUDEDIR} -c -o $@ lrsgmp.c
+	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -DMA -DGMP $(CPPFLAGS) -c -o $@ lrsgmp.c
 
 lrslib2-shr.o: lrslib.c lrslib.h
 	$(CC) ${CFLAGS} ${SHLIB_CFLAGS} -DMA -DSAFE -DB128 -DLRSLONG -c -o $@ lrslib.c
